homeassistant/components/zwave/__init__.py

```diff
@@ -410,28 +410,29 @@ def set_config_parameter(service):
         node = network.nodes[node_id]
         param = service.data.get(const.ATTR_CONFIG_PARAMETER)
         selection = service.data.get(const.ATTR_CONFIG_VALUE)
-        size = service.data.get(const.ATTR_CONFIG_SIZE, 2)
-        i = 0
+        size = service.data.get(const.ATTR_CONFIG_SIZE)
+        param_set = False
         for value in (
                 node.get_values(class_id=const.COMMAND_CLASS_CONFIGURATION)
                 .values()):
-            if value.index == param and value.type == const.TYPE_LIST:
-                _LOGGER.debug("Values for parameter %s: %s", param,
-                              value.data_items)
-                i = len(value.data_items) - 1
-        if i == 0:
-            node.set_config_param(param, selection, size)
-        else:
-            if selection > i:
-                _LOGGER.error("Config parameter selection does not exist! "
-                              "Please check zwcfg_[home_id].xml in "
-                              "your homeassistant config directory. "
-                              "Available selections are 0 to %s", i)
-                return
+            if value.index != param:
+                continue
+            if value.type in [const.TYPE_LIST, const.TYPE_BOOL]:
+                param_set = True
+                value.data = selection
+                _LOGGER.info("Setting config list parameter %s on Node %s "
+                             "with selection %s", param, node_id,
+                             selection)
+                break
```

Comment:
You can just return here (and below) instead of break. Then you won't need param_set